servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/ServerListenContext.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.transport.api;
+
+/**
+ * Context for controlling listen behavior.
+ */
+public interface ServerListenContext {
+    /**
+     * Toggles the server's channel accepting connection ability.
+     * <p>
+     * Passing a {@code false} value, will stop accepting connections on the server channel, without affecting any other
+     * interaction to currently open child channels (i.e., reads / writes).
```

Comment:
Since this is `transport-api` module, let's reduce amount of details that are relevant only to netty implementation. Consider something that Java users will understand easier:
```
Toggles the server's ability to accept new connections.

Passing a {@code false} value will signal the server to stop accepting new connections. It won't affecting any other
interaction to currently open connections (i.e., reads / writes).
```